src/py/flwr/server/superlink/linkstate/linkstate_test.py

```diff
@@ -353,6 +353,43 @@ def test_store_and_delete_tasks(self) -> None:
         assert state.num_task_ins() == 2
         assert state.num_task_res() == 1
 
+    def test_get_task_ids_from_run_id(self) -> None:
+        """Test get_task_ids_from_run_id."""
+        # Prepare
+        state = self.state_factory()
+        node_id = state.create_node(1e3)
+        run_id_0 = state.create_run(None, None, "8g13kl7", {}, ConfigsRecord())
+        # Insert tasks with the same run_id
+        task_ins_0 = create_task_ins(
+            consumer_node_id=node_id, anonymous=False, run_id=run_id_0
+        )
+        task_ins_1 = create_task_ins(
+            consumer_node_id=node_id, anonymous=False, run_id=run_id_0
+        )
+        # Insert a task with a different run_id to ensure it does not appear in result
+        run_id_1 = state.create_run(None, None, "9f86d08", {}, ConfigsRecord())
+        task_ins_2 = create_task_ins(
+            consumer_node_id=node_id, anonymous=False, run_id=run_id_1
+        )
+
```

Comment:
Should we include a third run that doesn't have any `TaskIns` in it and verify that `get_task_ids_from_run_id` is working as expected?

Reply:
Good suggestion. Added one test for a non-existing `run_id`.